src/cursor/cur_hs.c

```diff
@@ -479,186 +507,287 @@ __curhs_next_visible(WT_SESSION_IMPL *session, WT_CURSOR_HS *hs_cursor)
     return (ret);
 }
 
+/*
+ * __wt_curhs_search_near_before --
+ *     Set the cursor position at the requested position or before it.
+ */
+int
+__wt_curhs_search_near_before(WT_SESSION_IMPL *session, WT_CURSOR *cursor)
+{
+    return (__curhs_search_near_helper(session, cursor, true));
+}
+
+/*
+ * __wt_curhs_search_near_after --
+ *     Set the cursor position at the requested position or after it.
+ */
+int
+__wt_curhs_search_near_after(WT_SESSION_IMPL *session, WT_CURSOR *cursor)
+{
+    return (__curhs_search_near_helper(session, cursor, false));
+}
+
+/*
+ * __curhs_search_near_helper --
+ *     Helper function to set the cursor position based on search criteria.
+ */
+static int
+__curhs_search_near_helper(WT_SESSION_IMPL *session, WT_CURSOR *cursor, bool before)
+{
+    WT_DECL_ITEM(srch_key);
+    WT_DECL_RET;
+    int cmp;
+
+    WT_ERR(__wt_scr_alloc(session, 0, &srch_key));
+    WT_ERR(__wt_buf_set(session, srch_key, cursor->key.data, cursor->key.size));
+    WT_ERR(cursor->search_near(cursor, &cmp));
+    if (before) {
+        /*
+         * If we want to land on a key that is smaller or equal to the specified key, keep walking
+         * backwards as there may be content inserted concurrently.
+         */
+        if (cmp > 0) {
+            while ((ret = cursor->prev(cursor)) == 0) {
+                WT_STAT_CONN_INCR(session, cursor_skip_hs_cur_position);
+                WT_STAT_DATA_INCR(session, cursor_skip_hs_cur_position);
+                WT_ERR(__wt_compare(session, NULL, &cursor->key, srch_key, &cmp));
+                /* We find a key that is smaller or equal to the specified key. */
```

Comment:
This comment wording can be improved to something like: "Have we found a key that is smaller or equal..."

Reply:
Same below.

Reply:
Fixed.